tests/Install-Scripts.Test/GivenThatIWantToInstallTheSdkFromAScript.cs

```diff
@@ -0,0 +1,303 @@
+        {
+            get
+            {
+                // Download SDK using branches as channels
+                foreach(var sdkBranchInfo in _sdkBranches)
+                {
+                    yield return new object?[]
+                    {
+                        sdkBranchInfo.branch,
+                        sdkBranchInfo.versionRegex,
+                        null,
+                        null,
+                    };
+                }
+
+                // Download runtimes using branches as channels
+                foreach(var runtimeBranchInfo in _runtimeBranches)
+                {
+                    foreach(var runtime in _runtimes)
+                    {
+                        yield return new object?[]
+                        {
+                            runtimeBranchInfo.branch,
+                            runtimeBranchInfo.versionRegex,
+                            runtime.runtime,
+                            runtime.runtimeIdentifier,
+                        };
+                    }
+                }
+
+                // Download SDK and runtimes from darc channels.
+                foreach(var channelInfo in _channels)
+                {
+                    yield return new object?[]
+                    {
+                        channelInfo.channel,
+                        channelInfo.versionRegex,
+                        null,
+                        null,
+                    };
+
+                    foreach (var runtimeInfoe in _runtimes)
+                    {
+                        yield return new object?[]
+                        {
+                            channelInfo.channel,
+                            channelInfo.versionRegex,
+                            runtimeInfoe.runtime,
+                            runtimeInfoe.runtimeIdentifier,
+                        };
+                    }    
+                }
+            }
+        }
+
+        /// <summary>
+        /// The directory that the install script will install the .NET into.
+        /// </summary>
+        private readonly string _sdkInstallationDirectory;
+
+        /// <summary>
+        /// Instantiates a GivenThatIWantToInstallTheSdkFromAScript instance.
+        /// </summary>
+        /// <remarks>This constructor is called once for each of the tests to run.</remarks>
+        public GivenThatIWantToInstallTheSdkFromAScript()
+        {
+            _sdkInstallationDirectory = Path.Combine(
+                Path.GetTempPath(),
+                "InstallScript-Tests");
+
+            // In case there are any files from previous runs, clean them up.
+            try
+            {
+                Directory.Delete(_sdkInstallationDirectory, true);
+            }
+            catch (DirectoryNotFoundException)
+            {
+                // This is expected. Ignore the exception.
+            }
+
+            Directory.CreateDirectory(_sdkInstallationDirectory);
```

Comment:
is it possible that 2 or more runs will be ever started in parallel?  
If so, then it would be better to create unique subfolder for each run.

Reply:
I tried to make them parallel. Looks like xunit doesn't even support it.

Reply:
TE tests are all running in parallel, so it's possible. Should we look into this?

Reply:
I was worried at first, but it looks like all the tests are running within 8 minutes. Its probably not worth investigating at this moment. However, I still randomized the path for each test case to make it ready to be parallelizable.